scripts/microbe_stage/species_system.lua

```diff
@@ -0,0 +1,211 @@
+
+--------------------------------------------------------------------------------
+-- SpeciesComponent
+--
+-- Holds information about an entity representing a species
+--------------------------------------------------------------------------------
+class 'SpeciesComponent' (Component)
+
+SPECIES_NUM = 0
+
+function SpeciesComponent:__init(name)
+    Component.__init(self)
+    self.num = SPECIES_NUM
+    SPECIES_NUM = SPECIES_NUM + 1
+    if name == nil then
+        self.name = "noname"..self.num
+    else
+        self.name = name
+    end
+    --self.name="Default" -- TODO TODODO TODODODO get names properly somehow
+    self.populationBonusFactor = 1.0
+    self.populationPenaltyFactor = 1.0
+    self.deathsPerTime = 1.0
+    self.birthsPerTime = 1.0
+    self.currentPopulation = 1
+    -- todo make these actually do stuff
```

Comment:
I felt bad for disabling all your population calculations, even though they were terrifically ad-hoc. This was a guilt-reminder to make populations calculations actually work, possibly using all those numbers. I guess what I should do is get rid of them entirely :P